src/lib.rs

```diff
@@ -219,12 +248,74 @@ impl Fernet {
         self._decrypt_at_time(token, ttl, current_time)
     }
 
+    #[cfg(not(feature = "rustcrypto"))]
+    fn _decrypt_at_time(
+        &self,
+        token: &str,
+        ttl: Option<u64>,
+        current_time: u64,
+    ) -> Result<Vec<u8>, DecryptionError> {
+        let (data, iv, rest) = Self::_decrypt_parse(token, ttl, current_time)?;
+        let ciphertext = &rest[..rest.len() - 32];
+        let hmac = &rest[rest.len() - 32..];
+
+        let hmac_pkey = openssl::pkey::PKey::hmac(&self.signing_key).unwrap();
+        let mut hmac_signer =
+            openssl::sign::Signer::new(openssl::hash::MessageDigest::sha256(), &hmac_pkey).unwrap();
+        hmac_signer
+            .update(&data.get_ref()[..data.get_ref().len() - 32])
+            .unwrap();
+        let expected_hmac = hmac_signer.sign_to_vec().unwrap();
+        if !openssl::memcmp::eq(&expected_hmac, hmac) {
+            return Err(DecryptionError);
+        }
+
+        let plaintext = openssl::symm::decrypt(
+            openssl::symm::Cipher::aes_128_cbc(),
+            &self.encryption_key,
+            Some(&iv),
+            ciphertext,
+        )
+        .map_err(|_| DecryptionError)?;
+
+        Ok(plaintext)
+    }
+
+    #[cfg(feature = "rustcrypto")]
     fn _decrypt_at_time(
         &self,
         token: &str,
         ttl: Option<u64>,
         current_time: u64,
     ) -> Result<Vec<u8>, DecryptionError> {
+        let (data, iv, rest) = Self::_decrypt_parse(token, ttl, current_time)?;
+        let ciphertext = &rest[..rest.len() - 32];
+        let hmac = &rest[rest.len() - 32..];
+
+        let mut mac = Hmac::<Sha256>::new_from_slice(&self.signing_key)
+            .expect("Signing key has unexpected size");
+        mac.update(&data.get_ref()[..data.get_ref().len() - 32]);
+
+        let expected_hmac = mac.finalize().into_bytes();
+
+        use subtle::ConstantTimeEq;
+        if hmac.ct_eq(&expected_hmac).unwrap_u8() == 0 {
```

Comment:
Clearer imo: `if !hmac.ct_eq(&expected_hmac).into() {`

Reply:
Good point. I had to give it a type hint when adding `!`, so it didn't get as clean, but still better than a random u8.